### PR TITLE
Fix some small bugs for Masks

### PR DESCRIPTION
**Pull request by JakeLin**
Fix some small bugs for Circle, Polygon, and Wave masks. Also, add examples for demonstrating all masks. 

@tbaranes Could you please have a look when you have time, thanks.

simulator screen shot jun 29 2016 11 21 33 pm
simulator screen shot jun 29 2016 11 21 36 pm

### CODE REVIEW

File: IBAnimatable/MaskDesignable.swift [hunk 1]
```diff
@@ -191,7 +196,13 @@ public extension MaskDesignable where Self: UIView {
   
   private func maskWaveFromString(mask: String) {
     let params = retrieveMaskParameters(mask, maskName: MaskType.Wave.rawValue).componentsSeparatedByString(",")
-    if let unwrappedWidth = Float(params[1]), unwrappedOffset = Float(params[2]) where params.count == 3 {
+    
+    guard params.count == 3 else {
```
Comment (tbaranes) [inline IBAnimatable/MaskDesignable.swift:200]: Not sure that is necessary, but shouldn't we able to use the wave with 1 or 2 parameters if we only need to set the direction... ? Anyway, that can be another PR.
That's comment is extendable to any masks using parameters.

**Comment by tbaranes** [comment]
Awesome, that's a really nice fix 👍

File: IBAnimatable/MaskDesignable.swift (hunk 1 above)
Comment (JakeLin) [inline IBAnimatable/MaskDesignable.swift:200]: Right, will fix it, I thought `if let unwrappedWidth = Float(params[1]), unwrappedOffset = Float(params[2]) where params.count == 3` should guard it, But it didn't, and crash the app.

Comment (JakeLin) [inline IBAnimatable/MaskDesignable.swift:200]: @tbaranes I have tested the other masks, they can support wrong number of parameters.
